internal/staging_lockfile/staging_lockfile.go

```diff
@@ -0,0 +1,173 @@
+package staging_lockfile
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/containers/storage/internal/rawfilelock"
+)
+
+// StagingLockFile represents a file lock used to coordinate access to staging areas.
+// Typical usage is via CreateAndLock (for a new file) or TryLockExisting (for an existing file),
+// both of which return a StagingLockFile that must eventually be released with UnlockAndDelete.
+// This ensures that access to the staging file is properly synchronized both within and across processes.
+//
+// WARNING: This struct MUST NOT be created manually. Use the provided helper functions instead.
+type StagingLockFile struct {
+	// The following fields are only set when constructing *StagingLockFile, and must never be modified afterwards.
+	// They are safe to access without any other locking.
+	file string
```

Comment:
Non-blocking:
```suggestion
	file string // Also the key in stagingLockFiles
```
or an equivalent documentation in for `stagingLockFiles` itself, wouldn’t hurt